plugins/inputs/s7comm/s7comm.go

```diff
@@ -348,6 +362,7 @@ func handleFieldAddress(address string) (*gos7.S7DataItem, converterFunc, error)
 	item := &gos7.S7DataItem{
 		Area:     area,
 		WordLen:  wordlen,
+		Bit:      extra,
```

Comment:
The field Bit will be ignored, unless `WordLen == s7wlbit` (=1). Including this doesn't hurt other queries, but feels like bad practice.
Alternatively the result for bit-queries could be returned after L333, but this obscures the progam flow, as the buffer is created afterwards.

Would be happy about some ideas, how to solve this more elegantly.